src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java

```diff
@@ -615,19 +851,59 @@ private static void updateRotation(final int rotation)
     }
 
     /**
-     * Rotate the structure counter clockwise.
+     * Action performed when rename button is clicked.
      */
-    private void rotateLeftClicked()
+    private void renameClicked()
     {
-        rotation = (rotation + ROTATE_LEFT) % POSSIBLE_ROTATIONS;
-        updateRotation(rotation);
+        final Structures.StructureName structureName = new Structures.StructureName(schematics.get(schematicsDropDownList.getSelectedIndex()));
+        @NotNull final WindowStructureNameEntry window = new WindowStructureNameEntry(structureName);
+        window.open();
     }
 
     /**
-     * Rotate the structure counter clockwise.
+     * Action performed when rename button is clicked.
      */
-    private static void mirror()
+    private void deleteClicked()
     {
-        Settings.instance.mirror();
+        confirmDeleteDialog = new DialogDoneCancel(getWindow());
+        confirmDeleteDialog.setHandler(this);
+        final Structures.StructureName structureName = new Structures.StructureName(schematics.get(schematicsDropDownList.getSelectedIndex()));
+        confirmDeleteDialog.setTitle(LanguageHandler.format("com.minecolonies.coremod.gui.structure.delete.title"));
+        confirmDeleteDialog.setTextContent(LanguageHandler.format("com.minecolonies.coremod.gui.structure.delete.body", structureName.toString()));
+        confirmDeleteDialog.open();
+    }
+
+    /**
+     * handle when a dialog is closed.
+     *
+     * @param dialog which is being closed.
+     * @param buttonId is the id of the button used to close the dialog.
+     */
+    public void onDialogClosed(final DialogDoneCancel dialog, final int buttonId)
+    {
+        if (dialog == confirmDeleteDialog && buttonId == DialogDoneCancel.DONE)
+        {
+            final Structures.StructureName structureName = new Structures.StructureName(schematics.get(schematicsDropDownList.getSelectedIndex()));
+            if (Structures.SCHEMATICS_SCAN.equals(structureName.getPrefix()))
+            {
+                if (Structures.deleteScannedStructure(structureName))
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)